src/main/java/core/basesyntax/service/strategy/SupplyOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    private FruitStorageDao storageDao;
+
+    public SupplyOperation(FruitStorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void handle(FruitTransaction transaction) {
+        storageDao.getAllFruits()
+                .merge(transaction.getFruit(), transaction.getQuantity(), Integer::sum);
+    }
```

Comment:
According to the checklist, we should not begin class or method implementation with an empty line. Please remove the redundant empty lines at the beginning of the class definition and the `handle` method.